File: script/dg_query.pl
#! /usr/bin/perl

# return couples sid state for all jobs idle and running
#$dir=".";
#$dg_log="dg.log";
#open (LOG, ">>$dir/$dg_log") || die "Unable to write to local log file $dir/$dg_log";
while ( <STDIN> ) {
    print STDERR;
}

$command = "edg-job-status -i ~/.bossEDGjobs -noint |";
open (QUERY , $command );
while ( <QUERY> ) {
    if ( $_ =~ /Status info for the Job :\s+(.*)\s*/) {
	$sid = $1;
	#print "1. SID= \"".$1."\"\n";
	$_ = <QUERY>;
	if ( $_ =~ /Current Status:\s+(\w+)\s*(\S*)\s*/ ) {
	   # print STDERR "$sid} <$1>\n";
	    $stat{$sid} =$1.$2;
	    }
    }
}
#print LOG "Dump current situation:\n";
foreach $sid ( keys %stat ) {
    #print STDERR "$sid $stat{$sid}\n";
    if      ( $stat{$sid} eq "Running" ) {
	print "$sid R\n";
    } elsif ( $stat{$sid} eq "Checkpointed" ) {
	print "$sid SC\n";
    } elsif ( $stat{$sid} eq "Scheduled" ) {
	print "$sid SS\n";
    } elsif ( $stat{$sid} eq "Ready" ) {
	print "$sid RE\n";
    } elsif ( $stat{$sid} eq "Waiting" ) {
	print "$sid SW\n";
    } elsif ( $stat{$sid} eq "Submitted" ) {
	print "$sid SU\n";
    } elsif ( $stat{$sid} eq "Undefined" ) {
	print "$sid UN\n";
    } elsif ( $stat{$sid} eq "Cancelled" ) {
	print "$sid SK\n";
    } elsif ( $stat{$sid} eq "Done(Success)" ) {
	print "$sid OR\n";
    } elsif ( $stat{$sid} eq "Aborted" ) {
	print "$sid SA\n";
    } elsif ( $stat{$sid} eq "Done(Aborted)" ) {
	print "$sid DA\n";
    } elsif ( $stat{$sid} eq "Cleared" ) {
	print "$sid SE\n";
    }
}
#print LOG "End dump durrent situation:\n";

File: script/dg_query.py
#!/usr/bin/env python
import sys
import os

import warnings
## Get rid of some useless warning
warnings.simplefilter("ignore", RuntimeWarning)
#
# Add EDG_WL_LOCATION to the python path
#
try:
    path = os.environ['EDG_WL_LOCATION']
except:
    msg = "Error: the EDG_WL_LOCATION variable is not set."
    sys.exit(1)
#
libPath=os.path.join(path, "lib")
sys.path.insert(0,libPath)
libPath=os.path.join(path, "lib", "python")
sys.path.append(libPath)
libPath=os.path.join(path, "bin")
sys.path.insert(0,libPath)
#
from edg_wl_userinterface_common_LbWrapper import Status
from Job import JobStatus
#
status = Status()
states = JobStatus.states
#
statusMap={'Undefined':'UN',
           'Submitted':'SU',
           'Waiting':'SW',
           'Ready':'SR',
           'Scheduled':'SS',
           'Running':'R',
           'Done':'SD',
           'Cleared':'SE',
           'Aborted':'SA',
           'Cancelled':'SK',
           'Unknown':'UN',
           'Done(failed)':'SA' 
           }
#
st=0
for line in sys.stdin:
    try:
        id = line.split('\n')[0]
        if len(id) == 0 :
            continue
        status.getStatus(id,0)
        err, apiMsg = status.get_error()
        if err:
            continue
        result=''
        dest_ce=''
        dest_ce_queue=''
        reason=''
        timestamp=''
        jobidInfo = status.loadStatus(st)
        try:
            result = jobidInfo[states.index('status')]
        except:
            continue
        try:
            reason = jobidInfo[states.index('reason')].replace(" ","-")
            reason = reason.replace("'","''")
        except :
            pass
        try:
            dest_ce = jobidInfo[states.index( 'destination' )].replace("https://", "")
            dest_ce_queue= (dest_ce.split(':')[1]).split("/")[1]
            dest_ce = dest_ce.split(':')[0]
        except :
            pass
        try:
            timestamp = jobidInfo[states.index('stateEnterTimes')]
            pos = timestamp.find(result)
            timestamp = timestamp[timestamp.find('=', pos)+1:timestamp.find(' ', pos)]
        except :
            pass
        try:
            if result == 'Done' and jobidInfo[ states.index('done_code') ] != '0' :
                result = 'Done(failed)'
        except :
            pass
        print id,statusMap[result],"SCHED_STATUS="+result,"DEST_CE="+dest_ce,"DEST_QUEUE="+dest_ce_queue,"STATUS_REASON="+reason,"LB_TIMESTAMP="+timestamp
        st=st+1
    except:
        continue

